regulations/download-files.py

```diff
@@ -0,0 +1,124 @@
+"""Download Regulations.gov documents from indexes containing document URLs"""
+
+import argparse
+import itertools
+import json
+import os
+
+import requests
+from tqdm.auto import tqdm
+
+from licensed_pile import logs
+
+
+def parse_args():
+    parser = argparse.ArgumentParser("Regulations.gov index builder")
+    parser.add_argument(
+        "--input-dir", required=True, help="Path to directory containing indexes"
+    )
+    parser.add_argument("--output-dir", required=True, help="Path to output directory")
+    parser.add_argument(
+        "--years",
+        nargs="+",
+        default=list(map(str, range(2000, 2024))),
+        help="Years of data to process",
+    )
+    parser.add_argument(
+        "--agencies",
+        nargs="+",
+        default=[
+            "bis",
+            "dot",
+            "epa",
+            "faa",
+            "fda",
+            "fema",
+            "ferc",
+            "fmcsa",
+            "fra",
+            "nhtsa",
+            "osha",
+            "phmsa",
+            "sec",
+            "uscg",
+        ],
+        help="Agencies to process",
+    )
+    parser.add_argument(
+        "--file-types",
+        nargs="+",
+        default=[".txt", ".htm", ".doc", ".docx"],
+        help="File types to download",
+    )
+    parser.add_argument(
+        "--overwrite",
+        action="store_true",
+        default=False,
+        help="Overwrite existing files in output directory",
+    )
+    return parser.parse_args()
+
+
+def download_file(url, filename):
+    try:
+        response = requests.get(url)
+    except:
+        return -1
+    if response.status_code == 200:
+        with open(filename, "wb") as file:
+            file.write(response.content)
+        return response.status_code
+    else:
+        return -1
+
+
+def main(args):
+    logger = logs.get_logger("regulations")
+    for year in args.years:
+        for agency in args.agencies:
+            os.makedirs(os.path.join(args.output_dir, year, agency), exist_ok=True)
+
+    for year, agency in itertools.product(args.years, args.agencies):
+        input_file = os.path.join(args.input_dir, year, f"{agency}.json")
+        output_dir = os.path.join(args.output_dir, year, agency)
+
+        if not os.path.exists(input_file):
+            logger.info(f"Skipping {agency} data from {year} -- no input file")
+            continue
+
+        logger.info(f"Processing {agency} data from {year}")
+
+        with open(input_file, "r") as f:
+            index = json.load(f)
+
+        pbar_stats = {
+            "Downloaded": 0,
+            "Errors": 0,
+            "Already Exists": 0,
+            "Wrong File Type": 0,
+        }
+        pbar = tqdm(index.items())
+        for doc_id, metadatas in pbar:
```

Comment:
Can you add a comment explaining this metadata format? It seems like each doc can have multiple metadata entries? And then each one is a different file type? Or can they overlap but we just use the first value (because the file existence check)?